base: Improve errors for failures to parse

bjackman opened the pull request:
Some events from the kernel cannot be parsed by old trace-cmd
binaries. This results in data_str starting with "[FAILED TO PARSE]",
which we ourselves fail to parse because we expect to find a
key=value pair.

This adds a missing 'continue' statement to avoid attempting to avoid
a KeyError when attempting to use an unset prev_key as a key into
data_dict, and also adds a special error for the "[FAILED TO PARSE]"
case.

valschneider commented on trappy/base.py:
```diff
@@ -206,6 +206,16 @@ def generate_data_dict(self, data_str):
+                        raise ValueError(
+                            'trace-cmd failed to parse the "{}" event. You may '
+                            'need to compile the latest trace-cmd and put it in '
+                            'your $PATH'.format(self.__class__.__name__))
```
I think `self.__class__.unique_word` would be more accurate as it will give the exact event name. Class name is already pretty good though, it'll be roughly the same thing but in CamelCase.

valschneider commented on trappy/base.py:
```diff
@@ -206,6 +206,16 @@ def generate_data_dict(self, data_str):
+                    else:
+                        raise ValueError(
+                            "TRAPpy's parser for '{}' failed to parse the line:"
+                            "\n{}".format(self.__clas__.__name__, data_str))
```
Ditto for event name

bjackman commented:
Phew.. I really really really don't want to break anything here so trying to be a bit hyper-cautious with testing.. added a unit test, I also added a separate exception class just because there was already one instance of that in the code.

bjackman commented:
OK, changed the FAILED TO PARSE case to just print a warning and continue. There might be missing data, but at least we know what went wrong and that the rest of the trace is probably fine.

valschneider commented on trappy/base.py:
```diff
@@ -206,6 +208,18 @@ def generate_data_dict(self, data_str):
+                if not prev_key:
+                    if 'FAILED TO PARSE' in data_str:
+                        warnings.warn(
+                            'trace-cmd failed to parse the "{}" event. You may '
```
Ideally we'd want to have each warning only once (one per failed to parse event). Can we just roll with the "warnings only printed once" thing or should we do something a bit better ?

bjackman replied:
I think it only prints it once for each event already

valschneider replied:
LGTM then
